### PR TITLE
DOC-608 | APM-660 | Release notes for memory accounting metrics and memory usage improvements

### PR DESCRIPTION
**Pull request by Simran-B**
#### Upstream PRs

- 3.10:
- 3.11:
- 3.12:

### CODE REVIEW

File: site/content/3.12/release-notes/version-3.12/incompatible-changes-in-3-12.md [hunk 1]
```diff
@@ -79,6 +79,26 @@ The second option is the recommended one, as it signals the intent more clearly,
...
+Due to the [improved memory accounting in v3.12](whats-new-in-3-12.md#improved-memory-accounting),
+certain AQL queries may now get aborted because they exceed the defined
+memory limit but didn't get killed in previous versions. This is because of the
+more accurate memory tracking that reports a higher (actual) usage now. It allows
+ArangoDB to more reliably detect and kill queries that go over the per-query and
+global query memory limit, potentially preventing out-of-memory crashes of
+_arangod_ processes.
+
+In particular, AQL queries that perform write operations now report a
+significantly higher `peakMemoryUsage` in cursors than before. This is also
```
Comment (jsteemann) [inline site/content/3.12/release-notes/version-3.12/incompatible-changes-in-3-12.md:93]: I find the "in cursors" unnecessary here and suggest to remove it.
```suggestion
significantly higher `peakMemoryUsage` than before. This is also
```

File: site/content/3.12/release-notes/version-3.12/whats-new-in-3-12.md [hunk 2]
```diff
@@ -54,6 +54,39 @@ for details.
...
+| Label | Description |
+|:------|:------------|
+| `arangodb_agency_node_memory_usage` | Memory used by Agency store/cache. |
+| `arangodb_index_estimates_memory_usage` | Total memory usage of all index selectivity estimates. |
+| `arangodb_internal_cluster_info_memory_usage` | Amount of memory spent in ClusterInfo. |
+| `arangodb_requests_memory_usage` | Memory consumed by incoming, queued, and currently processed requests. |
+| `arangodb_revision_tree_buffered_memory_usage` | Total memory usage of buffered updates for all revision trees. |
+| `arangodb_scheduler_queue_memory_usage` | Number of bytes allocated for tasks in the scheduler queue. |
+| `arangodb_scheduler_stack_memory_usage` | Approximate stack memory usage of worker threads. |
+| `arangodb_search_consolidations_memory_usage` | Current count of opened file descriptors for this index. |
```
Comment (jsteemann) [inline site/content/3.12/release-notes/version-3.12/whats-new-in-3-12.md:84]: This does not look accurate.

File: site/content/3.12/release-notes/version-3.12/whats-new-in-3-12.md [hunk 3]
```diff
@@ -54,6 +54,39 @@ for details.
...
+| `arangodb_search_mapped_memory` | Amount of memory in bytes that is mapped. |
```
Comment (jsteemann) [inline site/content/3.12/release-notes/version-3.12/whats-new-in-3-12.md:85]: Should be confirmed by others first.
```suggestion
| `arangodb_search_mapped_memory` | Amount of memory in bytes that is mapped for ArangoSearch views and indexes. |
```

File: site/content/3.12/release-notes/version-3.12/whats-new-in-3-12.md [hunk 4]
```diff
@@ -54,6 +54,39 @@ for details.
...
+| `arangodb_search_readers_memory_usage` | Amount of memory in bytes that is used for reading from this index. |
+| `arangodb_search_writers_memory_usage` | Amount of memory in bytes that is used for writing to this index. |
```
Comment (jsteemann) [inline site/content/3.12/release-notes/version-3.12/whats-new-in-3-12.md:87]: What does "this index" mean in this context?

**Review by nerpaula — APPROVED** [review]

**Pull request merged**